CrowServe/CrowServe.h

```diff
@@ -65,9 +71,53 @@ class Socket {
         deinit_sockets();
     };
 
-
     bool try_init();
-    void send();
+
+    template <typename Handler>
+    void listen(const Handler& handler) { // todo - include cv for cancel
+        m_thread = std::jthread{[this, handler] {
+            while (true) {
+                Header main_header{};
+                if (receive_into(main_header) < 1) {
+                    // TODO: error handling 0 = conn closed, -1 = check errorno
+                    std::cout << "error";
+                    return;
+                }
+                
+                for (u32 i = 0; i < main_header.message_count; i++) {
+                    MessageHeader message_header{};
+                    if (receive_into(message_header) < 1) {
+                        // TODO: error handling
+                        std::cout << "error";
+                        return;
+                    }
+
+                    std::span<const char> message;
+
+                    if (message_header.message_size > 0) {  
+                        char buffer[4096];
+                        auto bytes_received = recv(m_socket, buffer, message_header.message_size,0);
+                        if (bytes_received < 1) {
+                            // TODO: error handling
+                            std::cout << "error";
+                            return;
+                        }
+                        std::span<const char> message{buffer, (u32)bytes_received};
```

Comment:
This constructs a shadowing copy that gets destroyed on the next line (end of scope). `message = std::span<const char>{buffer, (u32)bytes_received};` must be used for the message to get overwritten.

Reply:
Good catch, fixed.